doc/tutorials/gpu-passthru.rst

```diff
@@ -0,0 +1,147 @@
+
+Pass-through GPU to Guest
+**************************
+
+1. Copy ``/usr/share/acrn/samples/nuc/launch_win.sh`` to ``install_win.sh``
+
+::
+
+   cp /usr/share/acrn/samples/nuc/launch_win.sh ~/install_win.sh
+
+2. Modify ``install_win.sh`` script to specify the Windows image you use.
+
+3. Modify ``install_win.sh`` script to enable GVT-d.
+
+Add the following commands before ``acrn-dm -A -m $mem_size -s 0:0,hostbridge \``
+
+::
+
+      gpudevice=`cat /sys/bus/pci/devices/0000:00:02.0/device`
+
+      echo "8086 $gpudevice" > /sys/bus/pci/drivers/pci-stub/new_id
+      echo "0000:00:02.0" > /sys/bus/pci/devices/0000:00:02.0/driver/unbind
+      echo "0000:00:02.0" > /sys/bus/pci/drivers/pci-stub/bind
+
+Replace ``-s 2,pci-gvt -G "$2" \`` with ``-s 2,passthru,0/2/0,gpu \``
+
+4. Run the ``launch_win.sh``.
+
+      .. note:: If you want to pass-through GPU to Clear Linux User VM, the steps are
+         the same as above except your script.
+
+Enable GVT-d GOP driver
+***********************
+
+Issue
+======
+When enabling GVT-d, guest OS couldn’t light up physical screen
+before OS driver load. So guest BIOS and
+grub GUI couldn’t be showed on the physical screen.
+
+Reason
+==========
+Physical display is initialized by GOP driver or VBIOS
+before OS driver load, but guest BIOS doesn’t have them.
+
+Solution
+==========
+Integrate GOP driver binary into OVMF as a Dxe driver,
+then guest OVMF could see GOP driver and run it in
+graphic pass-through environment.
+So physical display could be initialized
+by GOP and used by guest BIOS and guest grub.
+
+Steps
+======
+1. fetch ACRN OVMF
+
+::
+
+      git clone https://github.com/projectacrn/acrn-edk2.git
+
+2. fetch vbt and gop driver
+
+Fetch vbt and gop driver from the vendor according to your CPU model name.
+
+3. git apply the following two patches
+
+* `Use-the-default-vbt-released-with-GOP-driver.patch
+  <../_static/downloads/Use-the-default-vbt-released-with-GOP-driver.patch>`_
+* `Integrate-IntelGopDriver-into-OVMF.patch
+  <../_static/downloads/Integrate-IntelGopDriver-into-OVMF.patch>`_
+
+4. add vbt and gop driver into OVMF
+
+::
+
+      cp IntelGopDriver.efi  acrn-edk2/OvmfPkg/IntelGop/IntelGopDriver.efi
+      cp Vbt.bin acrn-edk2/OvmfPkg/Vbt/Vbt.bin
+
+5. compile OVMF
+
+::
+
+    cd edk2
+    git submodule update --init CryptoPkg/Library/OpensslLib/openssl
+
+    source edksetup.sh
+    make -C BaseTools
+
+    vim Conf/target.txt
+
+      ACTIVE_PLATFORM = OvmfPkg/OvmfPkgX64.dsc
+      TARGET_ARCH = X64
+      TOOL_CHAIN_TAG = GCC5
+
+    build -DFD_SIZE_2MB -DDEBUG_ON_SERIAL_PORT=TRUE
+
+Notes:
+
+   -  ``source edksetup.sh``, this step is needed for compilation every time
+      a shell is created.
+
+   -  This will generate the binary at
+      ``Build/OvmfX64/DEBUG_GCC5/FV/OVMF.fd``
+
+   -  Make sure your GCC is 5.X. GCC 6 and above is NOT supported.
+
+   -  Modify the launch script to specify the OVMF you built just now.
```

Comment:
Make it more explicit that the binary should be transfered to the target board (since it's not built there)

Reply:
Ok, I will add this step here.